sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/model/PluginContextImpl.java

```diff
@@ -63,23 +64,41 @@ public Plugin getPlugin(String pluginName) {
 
     @Override
     public <T> ServiceReference<T> publishService(Class<T> ifClass, T implObject) {
-        return registryService.publishService(ifClass, implObject,
+        return publishService(ifClass, implObject, StringUtils.EMPTY_STRING);
+    }
+
+    @Override
+    public <T> ServiceReference<T> publishService(Class<T> ifClass, T implObject, String uniqueId) {
+        return registryService.publishService(ifClass, implObject, uniqueId,
             new PluginServiceProvider(plugin));
     }
 
     @Override
     public <T> ServiceReference<T> referenceService(Class<T> ifClass) {
-        return registryService.referenceService(ifClass, pluginServiceFilter);
+        return referenceService(ifClass, StringUtils.EMPTY_STRING);
     }
 
     @Override
-    public <T> ServiceReference<T> referenceService(Class<T> ifClass, final String pluginName) {
-        return registryService.referenceService(ifClass, new PluginNameServiceFilter(pluginName));
+    public <T> ServiceReference<T> referenceService(Class<T> ifClass, String uniqueId) {
+        List<ServiceReference<T>> references = registryService
+            .referenceServices(new DefaultServiceFilter<T>()
+                .setProviderType(ServiceProviderType.ARK_PLUGIN).setServiceInterface(ifClass)
+                .setUniqueId(uniqueId));
+        return references.size() == 0 ? null : references.get(0);
     }
 
     @Override
-    public <T> List<ServiceReference<T>> referenceServices(Class<T> ifClass) {
-        return registryService.referenceServices(ifClass, pluginServiceFilter);
+    public List<ServiceReference> referenceServices(ServiceFilter serviceFilter) {
+        List<ServiceReference> references = registryService
+            .referenceServices(new DefaultServiceFilter()
+                .setProviderType(ServiceProviderType.ARK_PLUGIN));
+        List<ServiceReference> result = new ArrayList<>();
+        for (ServiceReference reference : references) {
+            if (serviceFilter.match(reference)) {
+                result.add(reference);
+            }
+        }
```

Comment:
I think this api is not good enough to use. It's a bit hard for user to referenceServices with serviceFilter. When I referenceServices, I first focus on interfaceType, then uniqueId and final the filter. Although filter contains the previous 2 elements, but it's not a friendly api from my opinion. And add `@suppressWarning("unchecked")` here